### ISSUE TITLE
Dependency composer

### ISSUE THREAD

**Issue by Gemorroj** [WsdlToPhp/PackageGenerator#95]
Why the project uses its own implementation of autoloader? from readme:
```php
require_once '/path/to/where/the/package/must/be/generated/vendor/autoload.php';
```
I can in my `composer.json` to indicate - `"classmap": ["/path/to/where/the/package/must/be/generated/ classmap"]`
Execute
```bash
composer dump-autoload
```
And use my global autoloader.

**Comment by mikaelcom** [comment]
To be more clear, it should be require_once `\_\_DIR\_\_ . '/vendor/autoload.php';` if the script is created within the directory where the package has been generated. It's a sample, it's as the user creates its own script. In fact, it should be as in the generated `tutorial.php` file.

The user has to customize the path to the autoload as in https://github.com/WsdlToPhp/PackageEws365/blob/develop/samples/calendar/FindItems.php.

This is not a specific implementation of autoloader in my point of view or at least it's just wrongly explained in the readme, don't you think?

**Comment by Gemorroj** [comment]
> This is not a specific implementation of autoloader in my point of view or at least it's just wrongly explained in the readme, don't you think?

Yeah, I guess.
I was confused by this line: https://github.com/WsdlToPhp/PackageGenerator/blob/develop/composer.json#L44
Why in the `composer` dependency?


Documentation is very comprehensive but difficult to read.
Installation: https://github.com/WsdlToPhp/PackageGenerator#programmatic (why do I need to clone the repository? But it doesn't say what you need to add `"wsdltophp/packagegenerator": "*"` in `composer.json`)
As an alternative example: https://github.com/wsdl2phpgenerator/wsdl2phpgenerator#installation
The library has less features, but simple and clear documentation. Probably, therefore, higher popularity.

**Comment by mikaelcom** [comment]
Composer is used for the [Composer file generation](https://github.com/WsdlToPhp/PackageGenerator/blob/develop/src/File/Composer.php).

The documentation is indeed long and possibly confusing. It has been completed over the time each time a new option has been added. Nowadays the documentation could possibly be enlightened in order to show the most common usage. So any specific needs could be exported into the FAQ. I take note of that and I'll try to propose an englightened version soon.

The last time I checked wsdl2phpgenerator, my project seemed more complete (from my point of view, not so objective) as it handles all the specifics of a WSDL, in addition the generated package seemed more complete and giving more features not only by the methods but by the fact that the generated classes inherit from generic classes that gives more flexibility and evolution. Its "popularity" is probably due to the fact that it exists since more than 10 years with several contributors whereas my project only exists since a little more that 2 years with only one main contributor 😉.

But any feedback is appreciated, I currently lack of ideas of improvements (as I also don't spend all my time using the project) that could be added to the generated classes. In addition, if any issue is detected in the generated package, feel free to open an issue, I often like fixing it quickly.
